packages/sdk/src/mb-impl/token.ts

```diff
@@ -0,0 +1,214 @@
+    network = Network.TESTNET,
+    ownerId,
+    metadata,
+  } = args;
+
+  const { symbol, icon, baseUri, reference, referenceHash } = metadata;
+
+  const data = {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    owner_id: ownerId,
+    metadata: {
+      spec: 'nft-1.0.0',
+      name: name.replace(/[^a-z0-9]+/gim, '').toLowerCase(),
+      symbol: symbol.replace(/[^a-z0-9]+/gim, '').toLowerCase(),
+      icon: icon ?? DEFAULT_MB_LOGO,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      base_uri: baseUri ?? null,
+      reference: reference ?? null,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      reference_hash: referenceHash ?? null,
+    },
+  };
+
+  return {
+    contractAddress: factoryContractId || getFactoryContract(network),
+    methodName: METHOD_NAMES.DEPLOY_TOKEN_CONTRACT,
+    args: data,
+  };
+};
+
+export const transferContractOwnership = (
+  args: TransferTokenContractOwnership,
+): TransactionArgs => {
+  const { nftContractId, nextOwner, options } = args;
+  const { keepMinters = true } = options;
+
+  return {
+    contractAddress: nftContractId,
+    args: {
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      new_owner: nextOwner,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      keep_old_minters: keepMinters,
+    },
+    methodName: METHOD_NAMES.TRANSFER_TOKEN_CONTRACT_OWNERSHIP,
+  };
+};
+
+export const mint = (args: MintArgs): TransactionArgs => {
+  const { nftContractId, options } = args;
+
+  return {
+    contractAddress: nftContractId,
+    args: {},
+    methodName: METHOD_NAMES.MINT,
+  };
+};
+
+// TODO: do we want this method? How can we reuse `mint` instead of having an extra method
+export const mintMore = (): void => {
+  return;
+};
+
+export const addMinter = (args: AddRemoveMinterArgs): TransactionArgs => {
+  const { minterId, nftContractId } = args;
+
+  return {
+    contractAddress: nftContractId,
+    args: {
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      account_id: minterId,
+    },
+    methodName: METHOD_NAMES.ADD_MINTER,
+  };
+};
+
+export const removeMinter = (args: AddRemoveMinterArgs): TransactionArgs => {
+  const { minterId, nftContractId } = args;
+
+  return {
+    contractAddress: nftContractId,
+    args: {
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      account_id: minterId,
+    },
+    methodName: METHOD_NAMES.REMOVE_MINTER,
+  };
+};
+
+export const batchChangeMinters = (
+  args: BatchChangeMinters,
+): TransactionArgs => {
+  const { addMinters, removeMinters, nftContractId } = args;
+
+  return {
+    contractAddress: nftContractId,
+    args: {
+      grant: addMinters.length > 0 ? addMinters : undefined,
+      revoke: removeMinters.length > 0 ? removeMinters : undefined,
+    },
+    methodName: METHOD_NAMES.BATCH_CHANGE_MINTERS,
+  };
+};
+
+export const revoke = (args: RevokeAccountArgs): TransactionArgs => {
```

Comment:
Would name it `revokeApprovals`, I thought "Why do we need another method to revoke minters"?

Also, shouldn't there be a `approve` for symmetry? `buy` could reuse that fn

Reply:
Perfect!